### ISSUE TITLE
"Failed to determine the rustc date"

### ISSUE THREAD

**Issue by sinetek** [PyO3/pyo3#929]
```
The minimum required Rust version is 1.42.0-nightly 2020-01-21.
```

I have rust, more recent than this. :

```
> rustc --version
rustc 1.43.1-nightly
```

I am getting the following error:

```
--- stderr
Error: "Failed to determine the rustc date"
```

### 🌍 Environment

 - Your operating system and version:
Gentoo Linux, as of May 15.

 - Your python version:
python 3.7

 - How did you install python (e.g. apt or pyenv)? Did you use a virtualenv?:
 - Your Rust version (`rustc --version`):
rustc 1.43.1-nightly

 - Your PyO3 version:
0.8.0

 - Have you tried using latest PyO3 master (replace `version = "0.x.y"` with `git = "https://github.com/PyO3/pyo3")?`:
Yes, same thing.

**Comment by gyakovlev** [comment]
gentoo rust maintainer here
we have an option to build rust releases with `channel=nightly`, this is such build, it's not the one from rustup.
It supports nightly features, so the date check  in build.rs needs to be somehow avoidable if the versions constraint is satisfied and channel is nightly.

**Comment by davidhewitt** [comment]
Hopefully within a few months we will support stable Rust so this check is unnecessary.

If you want to make a workaround for now, I would accept a PR which allowed setting an environment variable `PYO3_SKIP_VERSION_CHECK` to disable the check.

**Comment by davidhewitt** [comment]
The version check has now been removed from master and will be released soon with pyO3 `0.11`.
